### ISSUE TITLE
GPX contains duplicate location entry if via point specified

### ISSUE THREAD

**Issue by karussell** [graphhopper/graphhopper#323]
A minor 'issue' e.g. for [this route](https://graphhopper.com/maps/?point=51.343618%2C12.360772&point=51.344057%2C12.362323&point=51.345233%2C12.362055) if you do the GPX export you get a duplicate entry:

```xml
<trkpt lat="51.34403" lon="12.362339"><time>2015-02-11T08:19:26+00:00</time></trkpt>
<trkpt lat="51.34403" lon="12.362339"><time>2015-02-11T08:19:26+00:00</time></trkpt>
```

**Comment by sskkcisos** [comment]
@karussell , i am beginning to explore the codebase and as a first step tried my hands on this issue. I have attempted to fix this issue by not processing the stopover points(identified by equal lat,lon and nextlat,nexton) in the Instruction::fillGPXList function. Share your thoughts and suggestions.

**Comment by karussell** [comment]
> Share your thoughts and suggestions.

Hard without code :)

**Comment by sskkcisos** [comment]
Hi, this is the code change I am talking about 

```
        for (int i = 0; i < len; i++)
         {
-            list.add(new GPXEntry(lat, lon, ele, prevTime));
-
             boolean last = i + 1 == len;
             double nextLat = last ? nextInstr.getFirstLat() : points.getLatitude(i + 1);
             double nextLon = last ? nextInstr.getFirstLon() : points.getLongitude(i + 1);
+
+            // to handle via points, ignore stopovers
+            if(Double.compare(lat, nextLat) == 0 && Double.compare(lon, nextLon) == 0)
+            {
+                continue;
+            }
+
+            list.add(new GPXEntry(lat, lon, ele, prevTime));
```

**Comment by karussell** [comment]
Ah, you mean the stopover is identical to the first next point? Maybe we can remove the first next point at the source instead of hiding it?

**Comment by sskkcisos** [comment]
Yes but I believe the GPXExport uses the same route function that is used by the web UI as well.  If we remove the duplicate at the source, then maybe the stopover point won't be shown in the web UI. I have not validated the same, though. So I thought of removing it at the time of GPXExport. My thought was to make a change that does not affect the use of the duplicate(if any) at other modules.

**Comment by devemux86** [comment]
Essentially we have two instructions (via-point and next instruction) at the same coordinates.

Instead of one's complete removal, how about if we get one point but with both instructions merged?

**Comment by sskkcisos** [comment]
Will try it out. Seems like a cleaner solution than what I suggested.
